core/dbt/parser/schema_test_builders.py

```diff
@@ -241,7 +248,7 @@ def __init__(
 
         # use hashed name as alias if too long
         if compiled_name != fqn_name:
-            self.modifiers['alias'] = compiled_name
+            self.config['alias'] = compiled_name
```

Comment:
This is an inspired choice from #3316. Let's check to see if there's a user-supplied value of `config['alias']` before overriding it here.